### PR TITLE
Add leetcode gql schemas

### PR DESCRIPTION
**Pull request by akotadi**
# Description

Adding schemas to be used with [pydantic](https://docs.pydantic.dev/latest/) during the next PR, this will give structure to the graphql response received. 
Also updated discord.py version to 2.4.0 and fixed a test because of that update.

# Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

# How Has This Been Tested?

Not required, to be tested during the next PR when used.

# Checklist:

- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

**Comment by akotadi** [comment]
* **#85** <a href="https://app.graphite.dev/github/pr/Proyecto-Nutria/otter-welcome-buddy/85?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#83** <a href="https://app.graphite.dev/github/pr/Proyecto-Nutria/otter-welcome-buddy/83?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#82** <a href="https://app.graphite.dev/github/pr/Proyecto-Nutria/otter-welcome-buddy/82?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#84** <a href="https://app.graphite.dev/github/pr/Proyecto-Nutria/otter-welcome-buddy/84?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `develop`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @akotadi and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: otter_welcome_buddy/gql_service/models/gql_leetcode_model.py [hunk 1]
```diff
@@ -0,0 +1,706 @@
...
+    submit_stats_global: LeetcodeUserSubmitStatsModel | None = None
+    submit_stats: LeetcodeUserSubmitStatsModel | None = None
+    active_badge: LeetcodeUserBadgeModel | None = None
+    contest_badge: LeetcodeUserBadgeModel | None = None
+    tag_problem_counts: LeetcodeTagProblemCountsCategoryModel | None = None
+    language_problem_count: list[LeetcodeLanguageProblemCountModel] | None = None
+    is_discuss_admin: bool | None = None
+    is_discuss_staff: bool | None = None
+    is_active: bool | None = None
+    # nameColor - nullable
```
Comment (GoberInfinity) [inline otter_welcome_buddy/gql_service/models/gql_leetcode_model.py]: Consider removing the non used coded, or if its is going to be used in the future, comment why it is commented rn.

File: pyproject.toml [hunk 2]
```diff
@@ -6,14 +6,15 @@ authors = ["proyecto-nutria <contact@proyectonutria.com>"]
 
 [tool.poetry.dependencies]
 python = "^3.10"
-"discord.py" = "^2.0.1"
+"discord.py" = "2.4.0"
```
Comment (GoberInfinity) [inline pyproject.toml:9]: Should we stick to a fixed version of `discord.py`?

Comment (akotadi) [reply pyproject.toml:9]: Not sure, with the lock file should not be a problem, but when testing I noticed that this new version broke our tests. So my take will be let this as fixed version as is the "main  library", while we can let the others as now.

**Comment by akotadi** [comment]
### Merge activity

* **Oct 10, 10:47 PM EDT**: A user started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/Proyecto-Nutria/otter-welcome-buddy/82).
* **Oct 10, 10:49 PM EDT**: [Graphite](https://app.graphite.dev/github/pr/Proyecto-Nutria/otter-welcome-buddy/82) rebased this pull request as part of a merge.
* **Oct 10, 10:50 PM EDT**: A user merged this pull request with [Graphite](https://app.graphite.dev/github/pr/Proyecto-Nutria/otter-welcome-buddy/82).

**Pull request merged**